lua/gitlab/git.lua

```diff
@@ -16,9 +17,12 @@ local run_system = function(command)
 end
 
 ---Returns all branches for the current repository
+---@param args table|nil extra arguments for `git branch`
 ---@return string|nil, string|nil
-M.branches = function()
-  return run_system({ "git", "branch" })
+M.branches = function(args)
+  -- Load here to prevent loop
+  local u = require("gitlab.utils")
```

Comment:
Not sure if it is worth adding a `require_on_index` implementation to `gitlab.nvim`, as I am not sure how often this is actually an issue (require loops). But this can also be solved by adding something like:

```lua
M.require_on_index = function(require_path)
  return setmetatable({}, {
    __index = function(_, key)
      return require(require_path)[key]
    end,

    __newindex = function(_, key, value)
      require(require_path)[key] = value
    end,
  })
end
```

and then doing:

```lua
local u = require("gitlab.util.lazy").require_on_index("gitlab.utils")
```

And since it is not required until it is used, you do not run into the same require looping stuff that causes this to be necessary.